core/spec/models/spree/order_spec.rb

```diff
@@ -537,7 +537,7 @@ def compute(computable)
     let(:methods) { [ok_method, no_method] }
 
     it "does not include a payment method that is not suitable for this order" do
-      allow(Spree::PaymentMethod).to receive(:available_on_front_end).and_return(methods)
+      allow(Spree::PaymentMethod).to receive(:available_on_front_end, :within_transaction_limits).and_return(methods)
```

Comment:
well, i tried gabillion of different options, but never thought about that.


On 17.08.2016 17:11, Marta Wyszyńska wrote:
>
> In core/spec/models/spree/order_spec.rb 
> <https://github.com/spree/spree/pull/7536#discussion_r75096055>:
>
> > @@ -537,7 +537,7 @@ def compute(computable)
> >      let(:methods) { [ok_method, no_method] }
> >
> >      it "does not include a payment method that is not suitable for this order" do
> > -      allow(Spree::PaymentMethod).to receive(:available_on_front_end).and_return(methods)
> > +      allow(Spree::PaymentMethod).to receive(:available_on_front_end, :within_transaction_limits).and_return(methods)
>
> @krtschmr <https://github.com/krtschmr> I've investigated this failing 
> spec, and it seems that the |ok_method| double received the 
> |within_transactional_limits?| call, which the double was not 
> expecting. if you add the following in line 540: |allow(ok_method).to 
> receive(:within_transaction_limits?).with(order).and_return(true)| the 
> test will pass. :)
> The whole spec should look like this:
>
> |it "does not include a payment method that is not suitable for this 
> order" do allow(ok_method).to 
> receive(:within_transaction_limits?).with(order).and_return(true) 
> allow(Spree::PaymentMethod).to 
> receive(:available_on_front_end).and_return(methods) 
> expect(order.available_payment_methods).to match_array [ok_method] end |
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/spree/spree/pull/7536/files/10037714e11ac5e3cb3f238cc3dfeb5b3408bfb2#r75096055>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AMv1uhNtzqowGcjzsUL3H1JzZjmHGPkZks5qgt5BgaJpZM4Jkf12>.
>